### DISCUSSION TITLE
How to improve Listview performance when loading multiple items?

### DISCUSSION THREAD

**Discussion by cris-m** [dotnet/maui#11174] [Q&A] [answered]
I have a list of country `json` file which I load on a `Popup` `Listview `. The performance is not good. it takes time to load and the `Popup` does not show until all the data are load. The binding property of the list does not work when using complex `ViewCell`. How can I improve performance? I used the same technic in xamarin form but I was loading country data from `ISO`. Currently there is a problem with `ISO` in `iOS`. I decided to use `json` file with the list. It contains 244 countries. 

https://user-images.githubusercontent.com/29815096/200480397-1b2b1ddf-6d9b-4ead-8051-768c787920a7.mp4

**Comment by nicjay** [comment]
How long is the data request itself taking, ignoring UI? 
Does it make any difference if you try rendering something like only a `Label` per country in the `ViewCell`?

**Reply by cris-m** [reply]
Still delay but there is a time difference when loading just country name in `ViewCell` `Label`. The binding country name appear. But the `ListView` performance is not good. I can scroll to the end.


https://user-images.githubusercontent.com/29815096/200538409-72d7ca8a-97d8-45f9-a5bc-1dee5d3b6fee.mp4

**Comment by Vroomer** [comment, marked as answer]
Make sure that the ListView isn't in ScrollView or VerticalStackLayout - otherwise it might be forced to calculate height of all items. Also try setting HeightRequest for ListView and the layout of items. You might also try to swap ListView for CollectionView as that is a preferable control moving forward with better performance. If the items have the same height, set ItemSizingStrategy to MeasureFirstItem.

**Reply by cris-m** [reply]
` VerticalStackLayout` affects `ListView`. But there is a small delay rendering the `ItemTemplate` from a customer view.
